Core/clim-core/panes.lisp

```diff
@@ -2322,113 +2322,95 @@ SCROLLER-PANE appear on the ergonomic left hand side, or leave set to
-        (tw (text-size pane (clime:label-pane-label pane))))
-    (with-bounding-rectangle* (x1 y1 x2 y2) (sheet-region pane)
-      (multiple-value-bind (iright itop ileft ibottom
-                                   bright btop bleft bbottom)
+  (let* ((region (sheet-region pane))
+         (label (clime:label-pane-label pane))
+         (m0 2)
+         (a (text-style-ascent (pane-text-style pane) pane))
+         (d (text-style-descent (pane-text-style pane) pane))
+         (text-width (text-size pane label)))
+    (with-bounding-rectangle* (x1 y1 x2 y2) region
+      (multiple-value-bind (ileft itop iright ibottom
+                            text-x-offset text-y-offset
+                            border-margin)
           (label-pane-margins pane)
-        (declare (ignorable iright itop ileft ibottom))
+        (declare (ignore ileft itop iright ibottom))
         (multiple-value-bind (tx ty)
             (values (ecase (pane-align-x pane)
-                      (:left (+ x1 m0 (if (sheet-children pane)
-                                          (+ a m0 m0 d)
-                                          0)))
-                      (:right (- x2 m0 (if (sheet-children pane)
-                                           (+ a m0 m0 d)
-                                           0)
-                               tw))
-                      (:center (- (/ (- x2 x1) 2) (/ tw 2))))
+                      (:left (+ x1 text-x-offset))
+                      (:right (- x2 text-x-offset text-width))
+                      (:center (- (/ (- x2 x1) 2) (/ text-width 2))))
                     (ecase (label-pane-label-alignment pane)
-                      (:top (+ y1 m0 a))
+                      (:top (+ y1 text-y-offset))
                       (:bottom (- y2 m0 d))))
-	  (draw-rectangle* pane x1 (- ty a) x2 (+ ty d) :ink (pane-background pane))
-          (draw-text* pane (clime:label-pane-label pane)
-                      tx ty)
-          ;;;
+          ;; Draw label.
+          (draw-rectangle* pane x1 (- ty a) x2 (+ ty d) :ink (pane-background pane))
+          (draw-text* pane label tx ty)
+          ;; Draw border around child without drawing over the label
+          ;; text.
           (when (sheet-child pane)
-            (with-drawing-options (pane
-                                   :clipping-region
-                                   (region-difference
-                                    (sheet-region pane)
-                                    (make-rectangle* (- tx m0) (- ty a) (+ tx tw m0) (+ ty d))))
-              (draw-bordered-rectangle* pane (+ x1 bleft) (+ y1 btop) (- x2 bright) (- y2 bbottom)
-                                        :style :groove))))))))
+            (let ((text-region (make-rectangle* (- tx m0) (- ty a) (+ tx text-width m0) (+ ty d))))
+              (with-drawing-options (pane :clipping-region (region-difference
+                                                            region text-region))
```

Comment:
no need to break a line here anymore, eventually whole options clause could go into a separate line.

Reply:
Not sure what you mean. Like this

```lisp
(with-drawing-options
    (pane :clipping-region (region-difference region text-region))
```

?

Reply:
yes, that.

Reply:
Changed.